src/tilesource.js

```diff
@@ -275,28 +276,39 @@ $.TileSource.prototype = {
         return this._tileHeight;
     },
 
+    /**
+     * @function
+     * @param {Number} level
+     * @return {OpenSeadragon.TileSource} Chainable.
+     */
+    setMaxLevel: function( level ) {
+        this.maxLevel = level;
+        this._memoizeLevelScale( level );
+        return this;
+    },
+
     /**
      * @function
      * @param {Number} level
      */
     getLevelScale: function( level ) {
+        this._memoizeLevelScale( this.maxLevel );
+        return this.getLevelScale( level );
+    },
 
+    // private
+    _memoizeLevelScale: function( level ) {
```

Comment:
I don't think we need the flexibility of being able to pass in a `level`. I think it's probably better to stick with `this.maxLevel`.